### PR TITLE
Tests - HomeViewModel

### PR DESCRIPTION
**Pull request by victor-sarda**
## Description
This PR sets the ground foundation for unit tests in our project 👨‍🔬
I added a test scheme and enabled test coverage. I'm gonna split tests into multiple PRs, this one introduces `HomeViewModel` tests

⬇️ **Important changes** ⬇️
- Removed UI tests target (not needed for now)
- Refactored our `UserDefaults` utils to add a custom suite and make them testable
- Updated our `APIService` protocol to use `Result` as it's easier to test and recommended since it's native now
- Made our models conform to `Equatable`
- Custom `HomeCellStatsViewData` equality to make sure only the `String` in the attributed title is tested (`NSAttachment` makes the equality fail, I might change this to only pass a string in the view data and create the attributed one in the view)

## Attachments 
Our `HomeViewModel` is now 98% tests 🎉 , which brings the whole coverage in the code to 26% already

image

---
Linked to #59 & #50

### CODE REVIEW

File: ViteMaDose/Helpers/Utils/UserDefaultsUtils.swift [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+import Foundation
+
+extension UserDefaults {
+
+    // MARK: - Setup
+
+    static let userDefaultSuiteName = "app.vitemadose"
+
+    static var shared: UserDefaults {
```
Comment (yageek) [inline ViteMaDose/Helpers/Utils/UserDefaultsUtils.swift:16]: Using `var` on purpose here?

Comment (victor-sarda) [reply ViteMaDose/Helpers/Utils/UserDefaultsUtils.swift:16]: Nope! I'll change it 👍

**Review by pauljeannot — APPROVED** [review]
Good job again!

Only thing: you introduced `Result<>` in the APIService, but I wonder if it would not have been better to remove the Swift Package currently used, copy/paste its content and improve it to use Results directly inside. What do you think ?
Anyway, this is quite good like that and can be done later !

**Comment by victor-sarda** [comment]
> Good job again!
> 
> Only thing: you introduced `Result<>` in the APIService, but I wonder if it would not have been better to remove the Swift Package currently used, copy/paste its content and improve it to use Results directly inside. What do you think ?
> Anyway, this is quite good like that and can be done later !

Yes definitely! I did it because of time constraint (and it's a bit out of scope) but I'll add a new issue to remove this dependency 👍

**Pull request merged**